mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/definition/programmatic/impl/PropertyMappingVectorFieldOptionsStepImpl.java

```diff
@@ -36,14 +36,7 @@ public <F> IndexFieldTypeOptionsStep<?, F> initiate(IndexFieldTypeFactory factor
 						if ( dimension == null ) {
 							throw log.vectorDimensionNotSpecified();
 						}
-						// we consider arrays (byte[] and float[] in particular to be vector types, anything else will be
-						// treated as a non-vector field and delegated to a regular as(..) call.
-						if ( clazz.isArray() ) {
-							return factory.asVector( dimension, clazz );
-						}
-						else {
-							return factory.as( clazz );
-						}
+						return factory.asVector( dimension, clazz );
```

Comment:
if we get rid of this check then we are never getting to this check to fail:
https://github.com/hibernate/hibernate-search/blob/ce6e483c2322eee9a1cc8f2dbda73f3823035e94/mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/bridge/binding/impl/FieldModelContributorContextImpl.java#L82-L91

since we have only one vector-type interface, so no matter what we pass in to `factory#asVector(...)`  we are either getting a vector step (so not going to fail here) or we are passing something that will fail inside the factory itself with `log#cannotGuessVectorFieldType(..)`...

So it seems if we'd remove this check from here, then we'd need to update the `log#cannotGuessVectorFieldType` and all the info from the other message about using bridges and stuff... 

WDYT?

Reply:
I think the check could still fail with a custom binder that sets the field type to a non-vector type, no?
E.g. `@VectorField(valueBinder = @ValueBinderRef(type = MyIntegerBinder.class))`.

> So it seems if we'd remove this check from here, then we'd need to update the log#cannotGuessVectorFieldType and all the info from the other message about using bridges and stuff...

Can you point me to the test where you think the error message wouldn't be clear enough without this additional info?

Reply:
I see.

If you make the number of dimensions "optional" (i.e. set later through the DSL), we could in theory expose a `asAny(Class)` method in `IndexFieldTypeFactory` that would return a standard or vector interface, whatever makes sense, and then the check you commented on would be more useful. We'd probably have to rename`IndexFieldTypeFactory#as()` to `asStandard` to be consistent though...

But I'm honestly not sure this is worth your time or the annoyance that renaming `as()` will cause... Just forwarding the exception from the backend, which should say something similar to what the stub backend says, would be enough IMO.